src/main/scala/org/constellation/infrastructure/endpoints/middlewares/PeerAuthMiddleware.scala

```diff
@@ -52,22 +54,58 @@ object PeerAuthMiddleware {
     }
 
   def responseSignerMiddleware[F[_]: Sync](
-    privateKey: PrivateKey
+    privateKey: PrivateKey,
+    sessionTokenService: SessionTokenService[F]
   )(http: HttpRoutes[F])(implicit C: ContextShift[F]): HttpRoutes[F] =
     Kleisli { req: Request[F] =>
-      http(req).flatMap { res =>
-        new Http4sResponseSigner[F](getGenerator(privateKey))
-          .sign(res)
+      val logger = Slf4jLogger.getLogger[F]
+
+      for {
+        res <- http(req)
+        oHeaderToken <- sessionTokenService.getOwnToken
```

Comment:
can we don't use `oAbc` prefixes saying about type?

Reply:
But you see! Now that you read this code you immediately had a clue that it's an option ;). Haha, but no problem if it annoys you I'm perfectly fine without 'o' prefixes :).